blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java

```diff
@@ -172,6 +196,22 @@ private void publishEvent(Message message) {
         }
     }
 
+    @Extension
+    public static class InputStepPublisher implements PipelineInputStepListener {
+
+        @Override
+        public void onStepContinue(InputStep inputStep, WorkflowRun run) {
+            // fire an unpaused event in case the input step has received its input
+            try {
+                PubsubBus.getBus().publish(new RunMessage(run)
+                        .setEventName(Events.JobChannel.job_run_unpaused)
```

Comment:
might be more complicated than this, what if multiple stages are waiting for input?